pygdbmi/gdbmiparser.py

```diff
@@ -193,15 +193,20 @@ def assert_match(actual_char_or_str, expected_char_or_str):
 
 def _get_notify_msg_and_payload(result, stream: StringStream):
     """Get notify message and payload dict"""
-    token = stream.advance_past_chars(["=", "*"])
-    token = int(token) if token != "" else None
-    logger.debug("%s", fmt_green("parsing message"))
-    message = stream.advance_past_chars([","])
+    match = _GDB_MI_NOTIFY_RE.match(result)
+    groups = match.groups() if match else [""]
+    token = int(groups[0]) if groups[0] != "" else None
+    message = groups[1]
```

Comment:
How is this going to work if `match.groups()` is false? That means you assigned `[""]` to `groups` so `groups[1]` won't work.